test/e2e/vsphere_volume_ops_storm.go

```diff
@@ -0,0 +1,203 @@
+		10. Delete storage class.
+*/
+
+var _ = framework.KubeDescribe("vsphere volume operations storm [Volume]", func() {
+	f := framework.NewDefaultFramework("volume-ops-storm")
+	const DEFAULT_VOLUME_OPS_SCALE = 30
+	var (
+		client            clientset.Interface
+		namespace         string
+		storageclass      *storage.StorageClass
+		pvclaims          []*v1.PersistentVolumeClaim
+		persistentvolumes []*v1.PersistentVolume
+		err               error
+		volume_ops_scale  int
+		vsp               *vsphere.VSphere
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		if len(nodeList.Items) == 0 {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		if os.Getenv("VOLUME_OPS_SCALE") != "" {
+			volume_ops_scale, err = strconv.Atoi(os.Getenv("VOLUME_OPS_SCALE"))
+			Expect(err).NotTo(HaveOccurred())
+		} else {
+			volume_ops_scale = DEFAULT_VOLUME_OPS_SCALE
+		}
+		pvclaims = make([]*v1.PersistentVolumeClaim, volume_ops_scale)
+		vsp, err = vsphere.GetVSphere()
+		Expect(err).NotTo(HaveOccurred())
+	})
+	AfterEach(func() {
+		By("Deleting PVCs")
+		for _, claim := range pvclaims {
+			framework.DeletePersistentVolumeClaim(client, claim.Name, namespace)
+		}
+		By("Deleting StorageClass")
+		err = client.StorageV1beta1().StorageClasses().Delete(storageclass.Name, nil)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should create pod with many volumes and verify no attach call fails", func() {
+		By(fmt.Sprintf("Running test with VOLUME_OPS_SCALE: %v", volume_ops_scale))
+		By("Creating Storage Class")
+		scParameters := make(map[string]string)
+		scParameters["diskformat"] = "thin"
+		storageclass, err = client.StorageV1beta1().StorageClasses().Create(getVSphereStorageClassSpec("thinsc", scParameters))
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating PVCs using the Storage Class")
+		count := 0
+		for count < volume_ops_scale {
+			pvclaims[count] = framework.CreatePVC(client, namespace, getVSphereClaimSpecWithStorageClassAnnotation(namespace, storageclass))
+			count++
+		}
+
+		By("Waiting for all claims to be in bound phase")
+		persistentvolumes = waitForPVClaimBoundPhase(client, pvclaims)
+
+		By("Creating pod to attach PVs to the node")
+		pod := createPod(client, namespace, pvclaims)
+
+		By("Verify all volumes are accessible and available in the pod")
+		verifyVolumesAccessible(pod, persistentvolumes, vsp)
+
+		By("Deleting pod")
+		framework.DeletePodWithWait(f, client, pod)
+
+		By("Waiting for volumes to be detached from the node")
+		for _, pv := range persistentvolumes {
+			waitForVSphereDiskToDetach(vsp, pv.Spec.VsphereVolume.VolumePath, k8stype.NodeName(pod.Spec.NodeName))
+		}
+	})
+})
+
+// create pod with given claims
+func createPod(client clientset.Interface, namespace string, pvclaims []*v1.PersistentVolumeClaim) *v1.Pod {
```

Comment:
This func is more flexible than the existing `CreateClientPod` and `MakePod` in _framework/_pv_util.go_. I know you want this pr merged very soon, but what do you think about moving this func to _pv_util_ and fixing up the calls in _persistent_volumes.go_ and _persistent_volumes-vsphere.go_?